engine/verification/fetcher/engine.go

```diff
@@ -281,7 +334,16 @@ func (e Engine) validateStakedExecutionNodeAtBlockID(senderID flow.Identifier, b
 // through HandleChunkDataPack).
 func (e *Engine) NotifyChunkDataPackSealed(chunkID flow.Identifier) {
 	// we need to report that the job has been finished eventually
-	defer e.chunkConsumerNotifier.Notify(chunkID)
+	status, exists := e.pendingChunks.ByID(chunkID)
+	if !exists {
+		e.log.Debug().
+			Hex("chunk_id", logging.ID(chunkID)).
+			Msg("could not fetch pending status for sealed chunk from mempool, dropping chunk data")
+		return
+	}
+
+	e.chunkConsumerNotifier.Notify(status.ChunkLocatorID)
```

Comment:
better to wait until the pending chunk has been removed and then notify the consumer

Reply:
fixed in https://github.com/onflow/flow-go/pull/686/commits/b34e6d482fb4538cc7624033bcaf040d3fa7d45d